cdk_framework/EKS/lib/stacks/ec2-cluster-stack.ts

```diff
@@ -0,0 +1,94 @@
+    super(scope, id, props);
+
+    const logging = [
+      eks.ClusterLoggingTypes.API,
+      eks.ClusterLoggingTypes.AUDIT,
+      eks.ClusterLoggingTypes.AUTHENTICATOR,
+      eks.ClusterLoggingTypes.CONTROLLER_MANAGER,
+      eks.ClusterLoggingTypes.SCHEDULER,
+    ]
+
+    const ec2Instance = props.ec2_instance.toLowerCase()
+    const nodeSize = validateNodeSize(props.node_size, ec2Instance)
+
+    const workerRole = new Role(this, 'EKSWorkerRole', {
+      assumedBy: new ServicePrincipal('ec2.amazonaws.com'),
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName('AmazonPrometheusRemoteWriteAccess'),
+        ManagedPolicy.fromAwsManagedPolicyName('AWSXrayWriteOnlyAccess'),
+        ManagedPolicy.fromAwsManagedPolicyName('CloudWatchAgentAdminPolicy'),
+        ManagedPolicy.fromAwsManagedPolicyName('AmazonS3ReadOnlyAccess'),
+        ManagedPolicy.fromAwsManagedPolicyName('AWSAppMeshEnvoyAccess'),
+        ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSWorkerNodePolicy'),
+        ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2ContainerRegistryReadOnly'),
+        ManagedPolicy.fromAwsManagedPolicyName('AmazonEKS_CNI_Policy')
+      ]
+    });
+    this.cluster = new eks.Cluster(this, props.name, {
+    clusterName: props.name,
+    vpc: props.vpc,
+    vpcSubnets: [{subnetType: ec2.SubnetType.PUBLIC}],
+    defaultCapacity: 0,  // we want to manage capacity our selves
+    version: props.version,
+    clusterLogging: logging,
+    
+    });
+    this.cluster.addNodegroupCapacity('ng-' + ec2Instance, {
+        instanceTypes: [new ec2.InstanceType(ec2Instance + '.' + nodeSize)],
+        minSize: 2,
+        nodeRole: workerRole
+    })
+    this.cluster.awsAuth.addMastersRole(Role.fromRoleName(this, 'eks_admin_role', 'Admin'))
+
+  }
+}
+
+export interface EC2ClusterStackProps extends StackProps{
+    name: string;
+    vpc: Vpc;
+    version: KubernetesVersion;
+    ec2_instance: string;
+    node_size: string
+}
+
+
+function validateNodeSize(size: string, instance: string){
+    if(!size || size === null || size === 'null' || size === ''){
+        return null
+    }
+    const adjustedSize = size.toLowerCase()
+
+    if(instance === 't4g'){
+        if(!supportedT4gInstances.has(adjustedSize)){
+            throw new Error('Node size is not one of the options listed here https://www.amazonaws.cn/en/ec2/instance-types/')
+        }
+    } else {
+        if(!supportedNodeSizes.has(adjustedSize)){
+            throw new Error('Node size is not one of the options listed here https://www.amazonaws.cn/en/ec2/instance-types/')
+        }
+        if(instance === 'm5' && adjustedSize === 'medium'){
+            throw new Error('CPU architecture and node size are not compatible')
+        }
+        if(instance === 'm6g' && adjustedSize === '24xlarge'){
+            throw new Error('CPU architecture and node size are not compatible')
+        }
+    }
```

Comment:
nit: Checking that the node size is in the `supportedNodeSize` set and then using a switch statement may be easier to read here.

Reply:
I'm going to redo the validation because now I am using interfaces. So going to delete this function.

Reply:
The issue is that there are 2 sets because the node sizes for t4g is so different from the other ones, I decided to put it a whole different set. Let me know if you still think I should change it from this though.